posts/hello-world-in-kafka-using-python.md

```diff
@@ -0,0 +1,171 @@
+# Hello world in Kafka using Python
+
+Howdy! Welcome to blog series by [Timber.io](https://timber.io) on various tools and technologies.
+
+This blog is for you if you've,
+
+- Ever marvelled what is Kafka? 
+- Heard of streaming/queueing/messaging systems but wondering why should one use it? 
+- What are the true benefits of using them?
+- How would it fit with your current backend architecture? 
+- You might just be eager to get started with using Kafka! 
+
+<Illustration>
+
+_Just a disclaimer: we're a logging company here @ Timber. We'd love it if you tried out our product (it's seriously great!), but that's all we're going to advertise our product ... you guys came here to learn about Kafka and this guide won't disappoint._
+
+## What is this blog about ? 
+You might have got a hint of it already. This blog is about understanding what is Kafka, apprehending the need for a tool like Kafka and then getting started with it using Python. We're believers that the best way to learn something is to do it, so get out your terminal and your favorite code editor and get ready. 
+
+---
+## What is Kafka? Why should one use it?
+In short, Kafka is a distributed streaming platform.
+
+> Oh wait! What does that even mean?
+
+Imagine that you have a simple web application which consists of an interative UI, a web server and a database. 
+
+![Simple Architecture](../images/kafka-image-one.jpg)
+
+You need to record all the events such as clicks, requests, impressions and searches that take place on your web application and store them for computation, reporting and analysis, each of which is done by separate applications or services. A simple solution would be to store the data in your database and connect all other applications and services to your database.
+
+![Simple Architecture](../images/kafka-image-two.jpg)
+
+This might look simple, but you're not finished. There are multiple challenges that can arise. 
+
+1. Events like clicks, requests, impressions and searches results in high frequency interaction/requests or data flow to your webserver and your primary database may not be equipped to scale seamlessly. This could introduce high latencies as more and more events pour in to the server.
+2. If you go ahead and store high frequency data in database systems like SQL or Mongo, it would be hard to introduce and reconstruct a new system or a database on all of the historical data. You lose the flexibility to extend the capabilities of your system by introducing new techologies.
+3. What if you have data processing systems in place to process these events to gain deeper insights? Since these systems wouldn't be capable of handling high frequency reads and you wouldn't have access to the true source of data it is practically impossible to experiment with various data processing, machine learning algorithms on all of the data.
```

Comment:
Explain what it means to be a _true source of data_? I saw later in the post you wrote that we 'have all the data from all different systems residing at a single place making Kafka a true source of data', but why can't SQL DB's that link to each other also be a true source of data?